lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -217,8 +217,13 @@ void HWMemSimImpl::generateMemory(HWModuleOp op, FirMemory mem) {
   auto dataType = b.getIntegerType(mem.dataWidth);
 
   // Create registers for the memory.
-  Value reg = b.create<sv::RegOp>(UnpackedArrayType::get(dataType, mem.depth),
-                                  b.getStringAttr("Memory"));
+  sv::RegOp reg = b.create<sv::RegOp>(
+      UnpackedArrayType::get(dataType, mem.depth), b.getStringAttr("Memory"));
+
+  circt::sv::setSVAttributes(
+      reg,
+      sv::SVAttributesAttr::get(
+          b.getContext(), {std::make_pair("ram_style", R"("distributed")")}));
```

Comment:
I _think_ we only want to do this if the read latency is zero, correct?

Reply:
That makes sense to me, I added a condition.